[download fail]: bilibili ep can't dl

<!-- 请输入网站的名称 -->
<!-- enter the website name -->

**Website name**: bilibili

<!-- 你的操作系统 -->
<!-- your operating system -->

**OS:**: Windows10

<!-- 视频链接地址 -->
<!-- the video url -->

**Video URL:**: [url](https://www.bilibili.com/bangumi/play/ep1665563)

<!-- 请输入下载时显示的错误信息 -->
<!-- enter the error message when downloading -->

**Stack overflow**

```
D:\swap\bili>lux -c b -i https://www.bilibili.com/bangumi/play/ep1665563
panic: runtime error: index out of range [1] with length 0

goroutine 1 [running]:
github.com/iawia002/lux/extractors/bilibili.extractBangumi({0xc000038390, 0x2f}, {0xc000f04000, 0x2f71fa}, {0x0, {0x0, 0x0}, 0x1, 0x0, 0x1, ...})
        /home/runner/work/lux/lux/extractors/bilibili/bilibili.go:99 +0xf12
github.com/iawia002/lux/extractors/bilibili.(*extractor).Extract(0xf0150f?, {0xc000038390, 0x2f}, {0x0, {0x0, 0x0}, 0x1, 0x0, 0x1, {0x21f98b0, ...}, ...})
        /home/runner/work/lux/lux/extractors/bilibili/bilibili.go:362 +0x278
github.com/iawia002/lux/extractors.Extract({0xc000038390?, 0x0?}, {0x0, {0x0, 0x0}, 0x1, 0x0, 0xa, {0x21f98b0, 0x1}, ...})
        /home/runner/work/lux/lux/extractors/extractors.go:54 +0x351
github.com/iawia002/lux/app.download(0xc00040ce40, {0xc000038390, 0x2f})
        /home/runner/work/lux/lux/app/app.go:275 +0x2e8
github.com/iawia002/lux/app.New.func1(0xc00040ce40)
        /home/runner/work/lux/lux/app/app.go:252 +0x691
github.com/urfave/cli/v2.(*App).RunContext(0xc000754340, {0x1b87800, 0x2508560}, {0xc0000e6000, 0x5, 0x8})
        /home/runner/go/pkg/mod/github.com/urfave/cli/v2@v2.6.0/app.go:322 +0x928
github.com/urfave/cli/v2.(*App).Run(0x70?, {0xc0000e6000?, 0x25b9a98ae28?, 0x10000000010?})
        /home/runner/go/pkg/mod/github.com/urfave/cli/v2@v2.6.0/app.go:224 +0x2f
main.main()
        /home/runner/work/lux/lux/main.go:13 +0x39

D:\swap\bili>lux --version

lux: version 0.24.1, A fast and simple video downloader.
```

<!-- 如果能提供截图，则对于解决你的问题非常有帮助 -->
<!-- If you can provide screenshots, it will be very helpful to solve your problem. -->

**Screenshots**

none

<!-- 其他信息 -->
<!-- add any other context about the problem here -->

**Additional context**

none

--- a/extractors/bilibili/bilibili.go
+++ b/extractors/bilibili/bilibili.go
@@ -95,7 +95,7 @@ type bilibiliOptions struct {
 }
 
 func extractBangumi(url, html string, extractOption extractors.Options) ([]*extractors.Data, error) {
-	dataString := utils.MatchOneOf(html, `<script\s+id="__NEXT_DATA__"\s+type="application/json"\s*>(.*?)</script\s*>`)[1]
+	dataString := utils.MatchOneOf(html, `const playurlSSRData = ({[\s\S]+})`)[1]
 	epArrayString := utils.MatchOneOf(dataString, `"episode_info"\s*:\s*(.+?)\s*,\s*"season_info"`)[1]
 	fullVideoIdString := utils.MatchOneOf(dataString, `"videoId"\s*:\s*"(ep|ss)(\d+)"`)
 	epSsString := fullVideoIdString[1] // "ep" or "ss"

